File: hackerrank/practice/grid-challenge.cpp
#include <iostream>
#include <algorithm>

using namespace std;

char **a;

string solve(){
	int n;
	cin >> n;
	a = new char*[n];
	for (int i = 0; i < n; i+= 1){
		a[i] = new char[n];
	}
	char x;
	for (int i = 0; i < n; i += 1){
		for (int j = 0; j < n; j += 1){
			cin >> x;
			a[i][j] = x;
		}
	}
	for (int i = 0; i < n; i += 1){
		sort(a[i],a[i]+n);
	}
	// for (int i = 0; i < n; i += 1){
	// 	for (int j = 0; j < n; j += 1){
	// 		cout << a[i][j] << " ";
	// 	}
	// 	cout << endl;
	// }
	for (int i = 0; i < n; i += 1){
		for (int j = 1; j < n; j += 1){
			if (a[j][i] < a[j-1][i]) return "NO";
		}
	}
	return "YES";

}
int main(void){
	int t;
	cin >> t;
	while (t--){
		cout << solve() << endl;
	}
}

File: hackerrank/practice/archive/grid-challenge.py
def solve(a):
	for i in xrange(len(a)):
		a[i].sort()
	for i in xrange(len(a)):
		for j in xrange(1,len(a)):
			if a[j][i] < a[j-1][i]:
				return False
	return True

for _ in xrange(input()):
	n = input()
	a=[]
	for i in xrange(n):
		a.append(list(raw_input()))
	print "YES" if solve(a) else "NO"